config/main.py

```diff
@@ -601,6 +617,7 @@ def _reset_failed_services():
         'telemetry'
     ]
 
+    _remove_invalid_services(services_to_reset)
```

Comment:
>_remove_invalid_services [](http://example.com/codeflow?start=4&length=24)

If a service is disabled in FEATURE table, we can still "reset failed", right?

Reply:
No, is there a case in which we would want to?

Reply:
I am thinking of below timeline:
1. a service became unhealthy and trapped in a failed status
2. user decided to disabled it in FEATURE table
3. user `config reload`

The expectation should be the "failed" status be "reset", like a fresh new service.

You state "No". Do you mean technically impossible?

Reply:
If user disables service in feature table and reloads config.db, hostcfgd will mask and disable the feature. Meaning the service is no longer loaded or active.

If we try to reset-failed the service, reset-failed command will exit with error "service not loaded". 

The fix is that if service is masked we do not want to run reset-failed or restart commands as those will return errors.